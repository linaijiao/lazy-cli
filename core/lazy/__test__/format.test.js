const createProject = require('@linaj/test-utils/createProject')

test('cjs-format-cli', async () => {
  const project = await createProject({ name: 'cjs-format-cli' })
  await project.write('index.js', 'export default 123')
  await project.run('lazy --format cjs')
  expect(project.files).toMatchSnapshot()
  expect(project.require('dist/index.js')).toEqual({ default: 123 })
})

test('cjs-format-config', async () => {
  const project = await createProject({ name: 'cjs-format-config' })
  await Promise.all([
    project.write('index.js', 'export default 123'),
    project.write('lazy.config.js', 'exports.output = { format: "cjs" }')
  ])
  await project.run('lazy')
  expect(project.files).toMatchSnapshot()
  expect(project.require('dist/index.js')).toEqual({ default: 123 })
})

test('umd-format-cli', async () => {
  const project = await createProject({ name: 'umd-format-cli' })
  await project.write('index.js', 'export default 123')
  await project.run('lazy --format umd --module-name Foo')
  expect(project.files).toMatchSnapshot()
  expect(project.require('dist/index.js')).toEqual({ default: 123 })
})

test('umd-format-config', async () => {
  const project = await createProject({ name: 'umd-format-config' })
  await Promise.all([
    project.write('index.js', 'export default 123'),
    project.write(
      'lazy.config.js',
      'exports.output = { format: "umd", moduleName: "Foo" }'
    )
  ])
  await project.run('lazy')
  expect(project.files).toMatchSnapshot()
  expect(project.require('dist/index.js')).toEqual({ default: 123 })
})
