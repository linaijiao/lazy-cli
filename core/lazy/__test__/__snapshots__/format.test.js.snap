// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cjs-format-cli 1`] = `
Array [
  "dist/index.js",
  "dist/index.js.map",
  "index.js",
]
`;

exports[`cjs-format-config 1`] = `
Array [
  "dist/index.js",
  "dist/index.js.map",
  "index.js",
  "lazy.config.js",
]
`;

exports[`umd-format-cli 1`] = `
Array [
  "dist/index.html",
  "dist/index.js",
  "dist/index.js.map",
  "index.js",
]
`;

exports[`umd-format-config 1`] = `
Array [
  "dist/index.html",
  "dist/index.js",
  "dist/index.js.map",
  "index.js",
  "lazy.config.js",
]
`;
